cmake/module/GetTargetInterface.cmake

```diff
@@ -32,7 +32,15 @@ endfunction()
 function(get_target_interface var config target property)
   get_target_property(result ${target} INTERFACE_${property})
   if(result)
-    evaluate_generator_expressions(result "${config}")
+    # The CMake FindThreads module uses generator expressions to conditionally apply
+    # the `-pthread` flag, avoiding its addition in cases such as CUDA or Swift, which
```

Comment:
Rather than documenting CUDA or Swift, which aren't relevant to this project, it'd be better to explain why this issue only occurs on some operating systems, like OpenBSD or FreeBSD, which we do support, which would make it clearer why we need to work around this.

Reply:
These systems' `libc`s do not implement `pthread` functionality and require linking to `libpthread` and the `-pthread` flag during compilation.

A workaround is required solely to evaluate generator expressions to make them readable in the summary. For an underlying build system they are evaluated automatically.

Reply:
Sure, if that's the issue then that's what should be documented, not something about CUDA & Swift.

Reply:
The comment has been rewritten.